docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,553 @@
+        break;
+      case TABLE_SCAN:
+        sb.append("TableScan");
+    }
+   ```
+
+2. Get the columns in projection and predicates.
+
+   ```java
+    if (!getFields().isEmpty()) {
+         sb.append(", Columns: ");
+         Joiner.on(", ").skipNulls().appendTo(sb, getFields());
+       }
+   ```
+
+3. Handled separately according to whether `isIndexScan` is true or not.
+
+   ```java
+      if (isIndexScan) {
+        sb.append(toIndexScanPhysicalPlan());
+      } else {
+        sb.append(toTableScanPhysicalPlan());
+      }
+   ```
+
+   - `toTableScanPhysicalPlan:`
+
+     Call `buildTableScan`, then call `toPhysicalPlan`.
+
+     ```java
+        sb.append("TableRangeScan");
+        sb.append(": {");
+        TiDAGRequest tableRangeScan = this.copy();
+        tableRangeScan.buildTableScan();
+        sb.append(tableRangeScan.toPhysicalPlan());
+     ```
+
+   - `toIndexScanPhysicalPlan:`
+
+      - First process `IndexRangeScan`, add the information of index scanned by `IndexRangeScan`, call `buildIndexScan`, then call `toPhysicalPlan`.
+      - if it is`DoubleRead`, add `TableRowIDScan`. first call `buildTableScan`, then call `toPhysicalPlan`.
+
+     ```java
+         sb.append("IndexRangeScan: ");
+         sb.append(index.colNames);
+         ...
+         TiDAGRequest indexRangeScan = this.copy();
+         indexRangeScan.buildIndexScan();
+         sb.append(indexRangeScan.toPhysicalPlan());
+         if (isDoubleRead()) {
+           sb.append(", TableRowIDScan:");
+           TiDAGRequest tableRowIDScan = this.copy();
+           tableRowIDScan.resetRanges();
+           tableRowIDScan.buildTableScan();
+           sb.append(tableRowIDScan.toPhysicalPlan());
+         }
+     ```
+
+   - `toPhysicalPlan`
+
+     Return `Range`, `Filters`, `Aggregates`, `GroupBy`, `OrderBy`,  `Limit`  to String.
+
+     ```java
+     if (!getRangesMaps().isEmpty()) {
+          sb.append(getRangeFliter())
+          sb.append(getRangesMaps())
+     }
+     if (!getPushDownFilters().isEmpty()) {
+         sb.append(getPushDownFilters())
+     }
+     if(!getPushDownAggregates().isEmpty()){
+         sb.append(getPushDownAggregates())
+     }
+     if (!getGroupByItems().isEmpty()) {
+          sb.append(getGroupByItems())
+     }
+     if (!getOrderByItems().isEmpty()) {
+         sb.append(getOrderByItems())
+     }
+     if (getLimit() != 0) {
+         sb.append(getLimit())
+     }
+     ```
+
+## Test Design
+
+1. Table without cluster index
+
+   ```SQL
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```sql
+   SELECT * FROM t1 where a>0 and b > 'aa'
+   ```
+
+2. Table with cluster index
+
+   ```sql
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) clustered
```

Comment:
> it is better to use a hint to avoid failure in TiDB 4.0

but in TiDB 6.0 the default primary key is not clustered, If we don`t use this, It may appear to be some problems when TiSpark runs with TiDB 6.0
![image](https://user-images.githubusercontent.com/38561029/174740593-e4a4e5af-4ba0-4c09-8af0-ada3e068b819.png)

Reply:
Add create table SQL for Tidb version smaller than 5.0